### PR TITLE
Miscellaneous plot fixes

### PR DESCRIPTION
**Pull request by philippjfr**
Smaller bug fixes along with tests to avoid regressions.

- [x] Fixes https://github.com/ioam/holoviews/issues/1757, https://github.com/ioam/holoviews/issues/1921 and https://github.com/ioam/holoviews/issues/2347

### CODE REVIEW

File: holoviews/plotting/bokeh/annotation.py [hunk 1]
```diff
@@ -35,9 +35,13 @@ def get_data(self, element, ranges, style):
...
         data['text'] = [element.text]
-        style['text_align'] = element.halign
+        if 'text_align' not in style:
+            style['text_align'] = element.halign
         baseline = 'middle' if element.valign == 'center' else element.valign
-        style['text_baseline'] = baseline
+        if 'text_baseline' not in style:
+            style['text_baseline'] = baseline
+        if 'text_font_size' not in style:
+            style['text_font_size'] = '%dPt' % element.fontsize
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/annotation.py]: One day we can try to allow the user to specify font sizes in something other than points. For now this is fine though...

**Comment by jlstevens** [comment]
Looks good. Happy to merge when the tests are green.

File: holoviews/plotting/bokeh/annotation.py (hunk 1 above)
Comment (philippjfr) [reply holoviews/plotting/bokeh/annotation.py]: Right, this is the semantics that matplotlib uses so I stuck with Pts. If you set the style option you can use a different unit.

Comment (jlstevens) [reply holoviews/plotting/bokeh/annotation.py]: To be honest using points as a default is probably fine and not worth changing (or even exposing). If anything the problem is documenting that the default is in points.

**Comment by jlstevens** [comment]
The pr build is now passing. Merging.

**Pull request merged**